### PR TITLE
C_CPP : Interpolation Search in Document Form

### PR DESCRIPTION
**Pull request by Pranshu321**
<hr>

### In this, I have covered all about the interpolation search that goes like this :
- What is interpolation Search
- Its Approach
- code and complexities.

Issue Number Closed:- #7122

<hr>

## Type of change 📝

- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] C/CPP

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Review by gunjan-g — APPROVED** [review]
@Saviour1001 @Suvraneel LGTM 🚀

File: C_CPP/Algorithmic Approaches/readme.md [hunk 1]
```diff
@@ -11,3 +11,4 @@
 - [Branch & Bound Algorithm](./Branch%20%26%20Bound)
 - [NP Complete Algorithm](./NP%20Completeness)
 - [Jump Search](./Algorithmic%20Approaches/Searching/jump_search.md)
+- [Interpolation Search](https://github.com/Pranshu321/winter-of-contributing/blob/patch-19/C_CPP/Algorithmic%20Approaches/Searching/Interpolation_Search.md)
```
Comment (Saviour1001) [inline C_CPP/Algorithmic Approaches/readme.md:14]: This link is to your file. Add a relative link instead. The syntax of relative link is 
```
- [My_Cool_Topic](./my_cool_file_name.md)
```

Comment (Suvraneel) [reply C_CPP/Algorithmic Approaches/readme.md:14]:
```suggestion
- [Interpolation Search](./Interpolation_Search.md)
```

**Review by Suvraneel — CHANGES_REQUESTED** [review]

File: C_CPP/Algorithmic Approaches/readme.md [hunk 2]
```diff
@@ -11,3 +11,4 @@
 - [Branch & Bound Algorithm](./Branch%20%26%20Bound)
 - [NP Complete Algorithm](./NP%20Completeness)
 - [Jump Search](./Algorithmic%20Approaches/Searching/jump_search.md)
+- [Interpolation Search](./Searching/Interpolation_Search.md)
```
Comment (Saviour1001) [inline C_CPP/Algorithmic Approaches/readme.md:14]:
```suggestion
- [Interpolation Search](./Interpolation_Search.md)
```

**Review by Suvraneel — CHANGES_REQUESTED** [review]
@Pranshu321  LGTM 🚀

**Pull request merged**